ui-tests/src/test/java/org/openmrs/reference/ManageProviderSchedulesTest.java

```diff
@@ -1,59 +1,64 @@
+ * Other possibilities are Rules or extending TestSetup. The latter possibilities might be applied
+ * in future if the test code become hard to maintain/read
  */
-public class ManageProviderSchedulesTest extends TestBase {
-    private HomePage homePage;
-    private HeaderPage headerPage;
-    private AppointmentBlocksPage appointmentBlocksPage;
-
-    @Before
-    public void setUp() throws Exception {
-        
-        homePage = new HomePage(page);
-        assertPage(homePage);
-        headerPage = new HeaderPage(driver);
-        appointmentBlocksPage = new AppointmentBlocksPage(page);
-    }
-
-    @Ignore //ignored due to RA-904
-    @Test
-    public void manageProviderSchedulesTest() throws Exception {
-        appointmentBlocksPage.goToAppointmentBlock();
-        appointmentBlocksPage.selectLocation("Isolation Ward");
-        appointmentBlocksPage.clickOnCurrentDay();
-        appointmentBlocksPage.selectLocationBlock("Isolation Ward");
-        appointmentBlocksPage.enterStartTime("08");
-        appointmentBlocksPage.enterService("gyne");
-        appointmentBlocksPage.enterProvider("Super User");
-        appointmentBlocksPage.clickOnSave();
-        assertNotNull("Gynecology", appointmentBlocksPage.CURRENT_DAY);
-        appointmentBlocksPage.findBlock();
-        appointmentBlocksPage.clickOnEdit();
-        appointmentBlocksPage.enterService("derma");
-        appointmentBlocksPage.clickOnSave();
-        assertNotNull("Dermatology", appointmentBlocksPage.CURRENT_DAY);
-        appointmentBlocksPage.findBlock();
-        appointmentBlocksPage.clickOnDelete();
-        appointmentBlocksPage.clickOnConfirmDelete();}
-
-    @After
-    public void tearDown() throws Exception {
-        headerPage.clickOnHomeIcon();
-        headerPage.logOut();
-    }
+public class ManageProviderSchedulesTest extends ReferenceApplicationTestBase {
+	
+	protected HomePage homePage;
+	
+	protected AppointmentBlocksPage appointmentBlocksPage;
+	
+	protected String locationName;
+	
+	protected String locationUuid;
+	
+	protected String provider;
+	
+	@Before
+	public void setUp() throws Exception {
+		homePage = new HomePage(page);
+		assertPage(homePage);
+		AppointmentSchedulingPage appointmentSchedulingPage = homePage.goToAppointmentScheduling();
+		appointmentBlocksPage = appointmentSchedulingPage.goToAppointmentBlock();
+		/*
+		 * TODO The following code should be re-nabled when bug AM-186 is fixed
+		 * Location might not be available when running this test, thus we create one
+		 * locationName = "Location-"+TestData.randomSuffix();
+		 * We must be sure that the location has been created, the test will fail otherwise
+		 * locationUuid = TestData.createLocation(locationName);
+		 * assertNotNull(locationUuid);
+		*/
+		locationName = "Isolation Ward";
+		provider = "Provider " + TestData.randomSuffix();
+	}
+	
+	@After
+	public void tearDown() throws Exception {
+		/*
+		 *  TODO delete the location. Re-enable when bug AM-186 is fixed.
```

Comment:
Well, I noticed that many tests miss them. 
I tried logOut and it didn't work, but I want to give another try to it.